pkg/datastore/postgres/repositories/checked_headers_repository_test.go

```diff
@@ -245,62 +273,90 @@ var _ = Describe("Checked Headers repository", func() {
 				}
 
-				nodeOneMissingHeaders, err := repo.UncheckedHeaders(startingBlockNumber, endingBlockNumber, uncheckedCheckCount)
+				nodeOneMissingHeaders, err := repo.UncheckedHeaders(firstBlock, thirdBlock, uncheckedCheckCount)
 				Expect(err).NotTo(HaveOccurred())
 				nodeOneHeaderBlockNumbers := getBlockNumbers(nodeOneMissingHeaders)
-				Expect(nodeOneHeaderBlockNumbers).To(ConsistOf(startingBlockNumber, middleBlockNumber, endingBlockNumber))
+				Expect(nodeOneHeaderBlockNumbers).To(ConsistOf(firstBlock, secondBlock, thirdBlock))
 
-				nodeTwoMissingHeaders, err := repoTwo.UncheckedHeaders(startingBlockNumber, endingBlockNumber+10, uncheckedCheckCount)
+				nodeTwoMissingHeaders, err := repoTwo.UncheckedHeaders(firstBlock, thirdBlock+10, uncheckedCheckCount)
 				Expect(err).NotTo(HaveOccurred())
 				nodeTwoHeaderBlockNumbers := getBlockNumbers(nodeTwoMissingHeaders)
-				Expect(nodeTwoHeaderBlockNumbers).To(ConsistOf(startingBlockNumber+10, middleBlockNumber+10, endingBlockNumber+10))
+				Expect(nodeTwoHeaderBlockNumbers).To(ConsistOf(firstBlock+10, secondBlock+10, thirdBlock+10))
 			})
 		})
 
 		Describe("when ending block is -1", func() {
 			var endingBlock = int64(-1)
 
 			It("includes all non-checked headers when ending block is -1 ", func() {
-				headers, err := repo.UncheckedHeaders(startingBlockNumber, endingBlock, uncheckedCheckCount)
+				headers, err := repo.UncheckedHeaders(firstBlock, endingBlock, uncheckedCheckCount)
 				Expect(err).NotTo(HaveOccurred())
 
 				headerBlockNumbers := getBlockNumbers(headers)
-				Expect(headerBlockNumbers).To(ConsistOf(startingBlockNumber, middleBlockNumber, endingBlockNumber, outOfRangeBlockNumber))
+				Expect(headerBlockNumbers).To(ConsistOf(firstBlock, secondBlock, thirdBlock, lastBlock))
 			})
 
 			It("excludes headers that have been checked more than the check count", func() {
 				_, err = db.Exec(`UPDATE public.headers SET check_count = 1 WHERE id = $1`, headerIDs[1])
 				Expect(err).NotTo(HaveOccurred())
 
-				headers, err := repo.UncheckedHeaders(startingBlockNumber, endingBlock, uncheckedCheckCount)
+				headers, err := repo.UncheckedHeaders(firstBlock, endingBlock, uncheckedCheckCount)
 				Expect(err).NotTo(HaveOccurred())
 
 				headerBlockNumbers := getBlockNumbers(headers)
-				Expect(headerBlockNumbers).To(ConsistOf(startingBlockNumber, endingBlockNumber, outOfRangeBlockNumber))
-				Expect(headerBlockNumbers).NotTo(ContainElement(middleBlockNumber))
+				Expect(headerBlockNumbers).To(ConsistOf(firstBlock, thirdBlock, lastBlock))
+				Expect(headerBlockNumbers).NotTo(ContainElement(secondBlock))
 			})
 
 			Describe("when header has already been checked", func() {
-				It("does not exclude header with block number > 15 back from latest", func() {
-					_, err = db.Exec(`UPDATE public.headers SET check_count = 1 WHERE id = $1`, middleBlockNumberID)
+				It("includes header with block number > 15 back from latest with check count of 1", func() {
```

Comment:
These tests go in pairs. The former demonstrates a header which is right at the threshold of being considered "unchecked", and the latter demonstrates that a header which is 1 block after the first one is _not_ considered "unchecked".